### ISSUE TITLE
Product-based proposals : Public Pool 

### ISSUE THREAD

**Issue by anthonyolazabal** [livepeer/Grant-Program#44]
**Give a 3 sentence description about this proposal.**
I currently operate a pool on the Chia and Chives blockchains (https://xchpool.org and our explorer : https://explorer.xchpool.org) and I'm already participating as an orchestrator and transcoder on the Livepeer network, I would like to develop a public Livepeer pool.

**Describe the problem you are solving.** 
Provide transcoding possibilities to GPUs miner with dual mining and to low budget stackers. We will provide the services, software and associated documentation.

**Describe the solution you are proposing.** 
The idea is to provide to transcoders and stackers a monitoring dashboard with their statistics and the overall statistics of the pool. Also provide an Android and iOS mobile application to monitor their participation and pool statistics. Automatic payment management.

**Describe the scope of the project including a rough timeline and milestones** 
The idea germinated a few weeks ago following a discussion with a colleague who is also an orchestrator on the Livepeer network.
We think that it's worth waiting for the L2 migration, but we can start the development quickly to have a MVP version in the next months. 

**Please estimate hours spent on the project based on the above** 
Before going further, I would like to know if having another public pool is interesting for you or not. We will basically need some technical support if we have questions regarding the protocol and the operations on the network. We take care of the hosting and high availability of the services that we provide. Once we have a feedback from you, we will build a project plan and some mockups for our platform. If you want to discuss it, you can contact me on Discord : LeBasque#7423
Thanks in advance for your feedback.

**Comment by nelsorya** [comment]
Hi @anthonyolazabal, thanks for putting this together. How many hours do you anticipate this grant would take to complete? Also, would you plan to make this open-source?

**Comment by anthonyolazabal** [comment]
Hello @nelsorya, as a high level estimation, between 200 and 400 hours of work to build all services. 
What do you mean by make this open-source ?

**Comment by nelsorya** [comment]
Would the code base for this be made open-source to the community or would it be kept closed-source?

**Comment by anthonyolazabal** [comment]
We can make a good part of the code public, but the most crtitical part (transcoding rewards for example) would remain private to avoid security issues.

**Comment by nelsorya** [comment]
@anthonyolazabal I'm afraid we are only issuing grants for Open Source Public Pools, given the existing for profit opportunity for closed-source public pools. If you were interested in exploring an entirely open-source alternative technical design that minimizes these security risks, this is something we could potentially issue a grant for.

**Comment by anthonyolazabal** [comment]
@nelsorya in that case let me rephrase my proposition. If you want to have a public pool code, I suggest we work together on two parts. We can initially work on the official implementation of a pool protocol directly in the livepeer program (I don't see anything regarding that in the code today), two simple aspects, the first one, for transcoders, we add an option (for example -poolTrsEthAdd <ETH address> ) allowing the participating transcoders to be identified for their work within the pool, and the second is to integrate within the code of the orchestrator the detection of this option to record locally in the DB the history of the processed jobs in order to be able to remunerate the transcoders fairly. Secondly, we will work on creating a template to deploy a Livepeer pool. We will make the code public under the MIT license. We could exchange directly via a chat to be more efficient in our discussion. Feel free to reach me on discord.
